apps/repo_poller/lib/repo_poller/poller_supervisor.ex

```diff
@@ -38,11 +38,40 @@ defmodule RepoPoller.PollerSupervisor do
     Supervisor.init(children, opts)
   end
 
-  @spec setup_repo(String.t(), keyword()) :: Repo.t()
-  defp setup_repo(url, tasks_attrs) do
+  def start_child(json_repo, "github") do
+    pool_id = Config.get_connection_pool_id()
+
+    %{
+      "repository_url" => url,
+      "polling_interval" => interval,
+      "id" => id
+    } = Poison.decode!(json_repo)
```

Comment:
Wouldn't this make more sense to pass as a Map (json_repo) - is it being called as a JSON argument elsewhere?

Reply:
I did that thinking about making a clear abstraction over JSON between Admin and Poller, that's why i `encode` the record in the admin and send a JSON message via RPC, but yeah i can send the record directly (it will contain some useless ecto attributes for poller tho, or maybe trim those attributes and just send a plain map, don't know, thoughts?)

Reply:
Personally I'd go with the map, adding guard clause to clarify required attributes